### DISCUSSION TITLE
Session error using Espresso driver

### DISCUSSION THREAD

**Discussion by 1alextest** [appium/appium#18799] [Q&A]
I am trying to automate some test for an android app. I am using WebdriverIO-Appium-Espresso driver.  I keep getting the error below. Does any have any idea of how to fix it?
 
 
 Desired capbilities:
 {
 platformName: "Android",
            "appium:platformVersion": "12.0",
            "appium:deviceName": "Pixel_6",
            "appium:app": "./app/android/this-isthe-app",
            "appium:automationName": "Espresso",
            "appium:showGradleLog" : true,
            "appium:appPackage":"com.app.using",
 }
 Dependencies: 
├── @wdio/appium-service@8.11.1
├── @wdio/cli@8.10.6
├── @wdio/local-runner@8.11.1
├── @wdio/mocha-framework@8.11.0
├── @wdio/spec-reporter@8.11.0
├── @wdio/types@8.10.4
└── appium-espresso-driver@2.23.2
 
 error i am getting:
 ERROR webdriver: Request failed with status 500 due to session not created: A new session could not be created. Details: Espresso server process has been unexpectedly terminated. Check the Appium server log and the logcat output for more details. Check https://github.com/appium/appium-espresso-driver#troubleshooting regarding advanced session startup troubleshooting.

**Comment by portyu9** [comment]
Hi @1alextest 

That message means the Espresso instrumentation process crashed; the actual cause will be in Android `logcat`.

First retry with a clean Espresso server build:

```js
capabilities: [{
  platformName: 'Android',
  'appium:automationName': 'Espresso',
  'appium:forceEspressoRebuild': true,
  'appium:showGradleLog': true,
}]
```

At the same time capture:

```bash
adb logcat
```

Look immediately before the Espresso process exits for `FATAL EXCEPTION`, `NoSuchMethodError`, `ClassNotFoundException`, or `Resources$NotFoundException`.

If it is an AndroidX/library version conflict, configure `appium:espressoBuildConfig` so the Espresso server dependencies match those used by your application.

Also make sure you are testing a compatible/debug APK and that the Espresso server and AUT can be signed with compatible signatures.

So the fix is **not increasing the session timeout**: rebuild the Espresso server, inspect the instrumentation crash in logcat, and align the conflicting AndroidX/dependency version reported there.

Please mark this answer as accepted if it helped, thank you.
